R/utils.R

```diff
@@ -124,79 +124,35 @@ panel_item <- function(title, ..., collapsed = TRUE, input_id = NULL) {
-    non_total_coln <- c(TRUE, !grepl("All Patients", names(content)))
     df <- as.data.frame(mf$strings[seq(nr_header + 1, nrow(mf$strings)), , drop = FALSE])
     header_df <- as.data.frame(mf$strings[seq_len(nr_header), , drop = FALSE])
 
-    ft <- flextable::flextable(df)
-    ft <- flextable::delete_part(ft, part = "header")
-    ft <- flextable::add_header(ft, values = header_df)
-
-    # Add titles
-    ft <- flextable::set_caption(ft, flextable::as_paragraph(
-      flextable::as_b(mf$main_title), "\n", paste(mf$subtitles, collapse = "\n")
-    ),
-    align_with_table = FALSE
-    )
-
-    merge_index_body <- get_merge_index(mf$spans[seq(nr_header + 1, nrow(mf$spans)), , drop = FALSE])
-    merge_index_header <- get_merge_index(mf$spans[seq_len(nr_header), , drop = FALSE])
-
-    ft <- merge_at_indice(ft, lst = merge_index_body, part = "body")
-    ft <- merge_at_indice(ft, lst = merge_index_header, part = "header")
-    ft <- flextable::align_text_col(ft, align = "center", header = TRUE)
-    ft <- flextable::align(ft, i = seq_len(nrow(content)), j = 1, align = "left")
-    ft <- padding_lst(ft, mf$row_info$indent)
-    ft <- flextable::padding(ft, padding.top = 1, padding.bottom = 1, part = "all")
-    ft <- flextable::autofit(ft, add_h = 0)
-
-    width_vector <- c(
-      dim(ft)$widths[1],
-      rep(sum(dim(ft)$widths[-1]), length(dim(ft)$widths) - 1) / (ncol(mf$strings) - 1)
-    )
-    ft <- flextable::width(ft, width = width_vector)
-    ft <- custom_theme(ft)
-
-    # Add footers
-    ft <- flextable::add_footer_lines(ft, flextable::as_paragraph(
-      flextable::as_chunk(mf$main_footer, props = flextable::fp_text_default(font.size = 8))
-    ))
-    if (length(mf$main_footer) > 0 && length(mf$prov_footer) > 0) ft <- flextable::add_footer_lines(ft, c("\n"))
-    ft <- flextable::add_footer_lines(ft, flextable::as_paragraph(
-      flextable::as_chunk(mf$prov_footer, props = flextable::fp_text_default(font.size = 8))
-    ))
+    ft <- rtables::df_to_tt(df)
+    rtables::main_title(ft) <- mf$main_title
+    rtables::subtitles(ft) <- mf$subtitles
+    rtables::main_footer(ft) <- mf$main_footer
+    rtables::prov_footer(ft) <- mf$prov_footer
+    rtables::header_section_div(ft) <- mf$header_section_div
+    content <- ft
+    ft <- rtables::tt_to_flextable(ft, total_width = c(grDevices::pdf.options()$width - 1))
   } else if (inherits(content, "data.frame")) {
-    ft <- flextable::flextable(content)
-    ft <- custom_theme(ft)
+    ft <- rtables::df_to_tt(content)
+    content <- ft
+    ft <- rtables::tt_to_flextable(ft)
   } else {
     stop(paste0("Unsupported class `(", format(class(content)), ")` when exporting table"))
   }
 
-  if (flextable::flextable_dim(ft)$widths > 10) {
-    pgwidth <- 10.5
-    width_vector <- dim(ft)$widths * pgwidth / flextable::flextable_dim(ft)$widths
-    ft <- flextable::width(ft, width = width_vector)
-  }
+  ft <- rtables::theme_docx_default(tt = content)(ft)
```

Comment:
If the theme is `NULL` you get something similar to the terminal print

Reply:
I have removed the default theme then as its not exposed for user to set.

Reply:
[613f4e8](https://github.com/insightsengineering/teal.reporter/pull/265/commits/613f4e81ab77aa8880eb48d061fa8e80e316171b)